src/main/java/org/apache/mesos/state/StateStoreUtils.java

```diff
@@ -1,13 +1,16 @@
 package org.apache.mesos.state;
 
 import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Utilities for implementations and users of {@link StateStore}.
  */
 public class StateStoreUtils {
```

Comment:
This seems to create an inconsistency in how scheduler state is stored:

If I want to set the frameworkID:

```
frameworkState.setFrameworkId(id)
```

But if I want to set the suppressed state:

```
StateStoreUtils.setSuppressed(frameworkState.getStateStore(), false);
```

These are both just two different properties of the scheduler, but now they have significantly different interfaces.  I'd rather just write:

```
frameworkState.setFrameworkId(id)
frameworkState.setSuppressed(false)
```

Can we instead make a concrete super class: `class SchedulerState` that implements `setSuppressed`, `getSuppressed`, `setFrameworkId`, etc?